### PR TITLE
use `Process` instead of `Thread` to speed up the `build-all` option

### PR DESCRIPTION
**Pull request by qianyi-sun**
Use `Process` to replace `Thread` to speed up the `scl scenario build-all --clean <scenario_dir>` runtime.
Before:
```
(.venv) sqy201x@Patrick-Desktop:~/SMARTS$ time scl scenario build-all --clean ./scenarios/intersections/
build-scenario ./scenarios/intersections//2lane
build-scenario ./scenarios/intersections//2lane_circle
build-scenario ./scenarios/intersections//4lane
build-scenario ./scenarios/intersections//6lane
build-scenario ./scenarios/intersections//4lane_t
Waiting on scenarios/intersections/2lane ...
Warning: Speed of turning connection 'edge-north-NS_0->edge-west-EW_0' reduced by 44.96 due to turning radius of 4.62 (length=5.00, angle=90.00).
Warning: Speed of turning connection 'edge-north-NS_2->edge-east-WE_2' reduced by 40.49 due to turning radius of 16.45 (length=20.48, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_0->edge-north-SN_0' reduced by 44.96 due to turning radius of 4.62 (length=5.00, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_2->edge-south-NS_2' reduced by 40.49 due to turning radius of 16.45 (length=20.48, angle=90.00).
Warning: Speed of turning connection 'edge-south-SN_0->edge-east-WE_0' reduced by 44.96 due to turning radius of 4.62 (length=5.00, angle=90.00).
Warning: 8 total messages of type: Speed of % connection '%' reduced by % due to turning radius of % (length=%, angle=%).
Warning: Speed of turning connection 'edge-north-NS_0->edge-west-EW_0' reduced by 41.67 due to turning radius of 12.63 (length=15.48, angle=90.00).
Warning: Speed of turning connection 'edge-north-NS_0->edge-east-WE_0' reduced by 40.45 due to turning radius of 16.57 (length=20.64, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_0->edge-north-SN_0' reduced by 41.67 due to turning radius of 12.63 (length=15.48, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_0->edge-south-NS_0' reduced by 40.45 due to turning radius of 16.57 (length=20.64, angle=90.00).
Warning: Speed of turning connection 'edge-south-SN_0->edge-east-WE_0' reduced by 41.67 due to turning radius of 12.63 (length=15.48, angle=90.00).
Warning: 8 total messages of type: Speed of % connection '%' reduced by % due to turning radius of % (length=%, angle=%).
Waiting on scenarios/intersections/2lane_circle ...
Waiting on scenarios/intersections/4lane ...
Waiting on scenarios/intersections/6lane ...
Waiting on scenarios/intersections/4lane_t ...

real    0m10.042s
user    0m13.173s
sys     0m12.675s
```
After:
```
(.venv) sqy201x@Patrick-Desktop:~/SMARTS$ time scl scenario build-all --clean ./scenarios/intersections/
build-scenario ./scenarios/intersections//2lane
build-scenario ./scenarios/intersections//2lane_circle
build-scenario ./scenarios/intersections//4lane
build-scenario ./scenarios/intersections//6lane
build-scenario ./scenarios/intersections//4lane_t
Waiting on ./scenarios/intersections//2lane ...
Warning: Speed of turning connection 'edge-north-NS_0->edge-west-EW_0' reduced by 44.96 due to turning radius of 4.62 (length=5.00, angle=90.00).
Warning: Speed of turning connection 'edge-north-NS_2->edge-east-WE_2' reduced by 40.49 due to turning radius of 16.45 (length=20.48, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_0->edge-north-SN_0' reduced by 44.96 due to turning radius of 4.62 (length=5.00, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_2->edge-south-NS_2' reduced by 40.49 due to turning radius of 16.45 (length=20.48, angle=90.00).
Warning: Speed of turning connection 'edge-south-SN_0->edge-east-WE_0' reduced by 44.96 due to turning radius of 4.62 (length=5.00, angle=90.00).
Warning: 8 total messages of type: Speed of % connection '%' reduced by % due to turning radius of % (length=%, angle=%).
Warning: Speed of turning connection 'edge-north-NS_0->edge-west-EW_0' reduced by 41.67 due to turning radius of 12.63 (length=15.48, angle=90.00).
Warning: Speed of turning connection 'edge-north-NS_0->edge-east-WE_0' reduced by 40.45 due to turning radius of 16.57 (length=20.64, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_0->edge-north-SN_0' reduced by 41.67 due to turning radius of 12.63 (length=15.48, angle=90.00).
Warning: Speed of turning connection 'edge-east-EW_0->edge-south-NS_0' reduced by 40.45 due to turning radius of 16.57 (length=20.64, angle=90.00).
Warning: Speed of turning connection 'edge-south-SN_0->edge-east-WE_0' reduced by 41.67 due to turning radius of 12.63 (length=15.48, angle=90.00).
Warning: 8 total messages of type: Speed of % connection '%' reduced by % due to turning radius of % (length=%, angle=%).
Waiting on ./scenarios/intersections//2lane_circle ...
Waiting on ./scenarios/intersections//4lane ...
Waiting on ./scenarios/intersections//6lane ...
Waiting on ./scenarios/intersections//4lane_t ...

real    0m5.897s
user    0m13.309s
sys     0m5.346s
```

### CODE REVIEW

**Comment by Gamenot** [comment]
I think we will want to update the `CHANGELOG`.

**Review by Gamenot — CHANGES_REQUESTED** [review]
I think we will want to update the `CHANGELOG`.

**Review by Gamenot — APPROVED** [review]
I think we could still improve this by doing the builds serially when only 1 cpu is available but I do not think that is needed.

**Review by sah-huawei — COMMENTED** [review]
Looks good.  I just recommend a minor tweak to the acquire/release pattern.

File: cli/studio.py [hunk 1]
```diff
@@ -86,9 +86,11 @@ def _build_single_scenario(clean: bool, allow_offset_map: bool, scenario: str):
 def _build_single_scenario_proc(
     clean: bool, allow_offset_map: bool, scenario: str, semaphore: Semaphore
 ):
-    semaphore.acquire()
-    _build_single_scenario(clean, allow_offset_map, scenario)
-    semaphore.release()
+    semaphore.lock()
```
Comment (sah-huawei) [inline cli/studio.py:89]: oops, sorry, I first wrote `lock()` by accident and then edited it to `acquire()`.  But you were too fast!  my bad! :)

Comment (qianyi-sun) [reply cli/studio.py]: fixed in the latest commit:) Thanks for these suggestions Steve!

**Review by sah-huawei — APPROVED** [review]
Good work.
You should restart the CI jobs as they're currently failing.  (Looks like it's for the same reasons as in #1230.)

**Pull request merged**